File: app/src/main/java/com/progix/fridgex/light/adapter/tips/TipListAdapter.java
package com.progix.fridgex.light.adapter.tips;

import static android.view.animation.AnimationUtils.loadAnimation;
import static com.progix.fridgex.light.data.DataArrays.adviceImages;

import android.content.Context;
import android.os.Bundle;
import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.navigation.NavController;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.progix.fridgex.light.R;

import java.util.ArrayList;

public class TipListAdapter extends RecyclerView.Adapter<TipListAdapter.ViewHolder> {

    Context context;
    ArrayList<Pair<Integer, String>> tipList;
    NavController navController;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_tip, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Glide.with(context).load(adviceImages.get(tipList.get(position).first - 1)).into(holder.image);
        holder.category.setText(tipList.get(position).second);
        holder.itemView.setOnClickListener(v -> {
            Bundle bundle = new Bundle();
            bundle.putInt("advice", position);
            navController.navigate(R.id.nav_tip_list, bundle, null);
        });
        setAnimation(holder.itemView, position);
    }

    @Override
    public int getItemCount() {
        return tipList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        //MaterialCardView  ;
        ImageView image;
        TextView category;
        //View  ;

        public ViewHolder(@NonNull View view) {
            super(view);
            image = view.findViewById(R.id.image);
            category = view.findViewById(R.id.category);
        }

        public void clearAnimation() {
            itemView.clearAnimation();
        }

    }

    private int lastPosition = -1;

    private void setAnimation(View viewToAnimate, Integer position) {
        Animation animation = loadAnimation(context, R.anim.item_animation_fall_down);
        viewToAnimate.startAnimation(animation);
        lastPosition = position;
    }

    @Override
    public boolean onFailedToRecycleView(@NonNull ViewHolder holder) {
        return true;
    }

    @Override
    public void onViewDetachedFromWindow(@NonNull ViewHolder holder) {
        holder.clearAnimation();
    }

}

File: app/src/main/java/com/progix/fridgex/light/adapter/tips/TipListAdapter.kt
package com.progix.fridgex.light.adapter.tips

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.Animation
import android.view.animation.AnimationUtils.loadAnimation
import android.widget.ImageView
import android.widget.TextView
import androidx.navigation.NavController
import androidx.navigation.fragment.FragmentNavigatorExtras
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.progix.fridgex.light.R
import com.progix.fridgex.light.data.DataArrays.adviceImages


class TipListAdapter(
    var context: Context,
    private var tipList: ArrayList<Pair<Int, String>>,
    private var navController: NavController
) : RecyclerView.Adapter<TipListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_tip, parent, false)
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Glide.with(context).load(adviceImages[tipList[position].first - 1]).into(holder.image)
        holder.category.text = tipList[position].second
        holder.itemView.setOnClickListener {
            val bundle = Bundle()
            bundle.putInt("advice", position)
            holder.image.transitionName = "advice$position"
            val extras = FragmentNavigatorExtras(
                holder.image to holder.image.transitionName
            )
            navController.navigate(R.id.nav_tip_list, bundle, null, extras)
        }
        setAnimation(holder.itemView, position)

    }

    override fun getItemCount(): Int {
        return tipList.size
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val image: ImageView = view.findViewById(R.id.image)
        val category: TextView = view.findViewById(R.id.category)

        fun clearAnimation() {
            itemView.clearAnimation()
        }

    }

    private var lastPosition = -1
    private fun setAnimation(viewToAnimate: View, position: Int) {
        val animation: Animation =
            loadAnimation(context, R.anim.item_animation_fall_down)
        viewToAnimate.startAnimation(animation)
        lastPosition = position
    }

    override fun onFailedToRecycleView(holder: ViewHolder): Boolean {
        return true
    }

    override fun onViewDetachedFromWindow(holder: ViewHolder) {
        holder.clearAnimation()
    }

}